com.unity.cinemachine/Editor/Editors/CinemachineTriggerActionEditor.cs

```diff
@@ -1,152 +1,111 @@
+#endif
+                if (string.IsNullOrEmpty(helpText))
                 {
-                    if (GetTargetComponent<Animator>(targetProp.objectReferenceValue) == null
-                        && GetTargetComponent<PlayableDirector>(targetProp.objectReferenceValue) == null)
+                    switch (action)
                     {
-                        EditorGUILayout.HelpBox("Target must have a PlayableDirector or Animator in order to Play/Stop", MessageType.Warning);
+                        default: break;
+                        case CinemachineTriggerAction.ActionSettings.ActionModes.PriorityBoost:
+                            if (!HasComponent<CinemachineVirtualCameraBase>(targetObject))
+                                helpText = "Target must be a Cinemachine Camera in order to boost priority";
+                            break;
+                        case CinemachineTriggerAction.ActionSettings.ActionModes.Enable:
+                        case CinemachineTriggerAction.ActionSettings.ActionModes.Disable:
+                            if (targetObject is not Behaviour)
+                                helpText = "Target must be a Behaviour in order to Enable/Disable";
+                            break;
                     }
                 }
-#endif
-                if (!isCustom && targetProp.objectReferenceValue == null)
-                    EditorGUILayout.HelpBox("No action will be taken because target is not valid", MessageType.Info);
-
-                EditorGUILayout.Space();
-                EditorGUILayout.LabelField("This event will be invoked.  Add calls to custom methods here:");
-                EditorGUILayout.PropertyField(property.FindPropertyRelative(() => m_Def.Event));
-            }
-            property.serializedObject.ApplyModifiedProperties();
-            return expanded;
-        }
+                helpMessage.text = helpText;
+                helpMessage.SetVisible(!string.IsNullOrEmpty(helpText));
+            });
 
-        static T GetTargetComponent<T>(UnityEngine.Object obj) where T : Behaviour
-        {
-            UnityEngine.Object currentTarget = obj;
-            if (currentTarget != null)
+            static bool HasComponent<T>(Object obj) where T : Component
             {
-                GameObject targetGameObject = currentTarget as GameObject;
-                Behaviour targetBehaviour = currentTarget as Behaviour;
-                if (targetBehaviour != null)
-                    targetGameObject = targetBehaviour.gameObject;
-                if (targetBehaviour is T)
-                    return targetBehaviour as T;
-                if (targetGameObject != null && targetGameObject.TryGetComponent<T>(out var t))
-                    return t;
+                if (obj is T)
+                    return true;
+                var go = obj as GameObject;
+                if (go == null && obj is Component)
+                    go = (obj as Component).gameObject;
```

Comment:
You can cast directly because we know obj is Component here.
```
if (go == null && obj is Component)
     go = ((Component)obj).gameObject;
```
or
```
 if (go == null && obj is Component component)
     go = component.gameObject;
```

Reply:
> You can cast directly because we know obj is Component here.
I don't understand why that is any better than what is written.  Didn't we discuss this already?

Reply:
That was slightly different, I think. 

I think it was about using `as` instead of casting directly, but when `is` is used, then the situation is different.

Reply:
You didn't answer the real question, which was implied by: "I don't understand why that is any better than what is written."